tests/rptest/tests/prefix_truncate_test.py

```diff
@@ -0,0 +1,201 @@
+        return topics_info[0]
+
+    @cluster(num_nodes=5)
+    def test_prefix_truncate(self):
+        """
+        Test truncating at first segment
+
+        The parameters given in this test will likely remove no data but this
+        test is important reguardless. This is because the truncation will occur
+        in the first segment of the topic/partition which will only have an
+        effect of manipulating the start offset delta, which is the condition
+        desired to be tested.
+        """
+        num_records = 10240
+        records_size = 512
+        truncate_offset = 100
+
+        # Produce some data, wait for it all to arrive
+        kafka_tools = KafkaCliTools(self.redpanda)
+        kafka_tools.produce(self.topic, num_records, records_size, acks=1)
+        self._consume_and_count_bytes(0, num_records)
+
+        # Perform truncation at offset 100
+        low_watermark = self.delete_records(self.topic, 0, truncate_offset)
+        assert low_watermark == truncate_offset, f"Expected low watermark: {truncate_offset} observed: {low_watermark}"
+
+        # Consume up until end and assert final batch is the expected offset
+        self._consume_and_count_bytes(truncate_offset,
+                                      num_records - truncate_offset)
+        # Attempt to read one past known end
+        try:
+            self._consume_and_count_bytes(num_records, 1, timeout=2)
+            assert False, "Successfully read past known end"
+        except Exception as _:
+            pass
+
+        # Assert correctness of start & end offsets in topic metadata
+        topic_info = self._get_topic_info()
+        assert topic_info.id == 0, f"Partition id: {topic_info.id}"
+        assert topic_info.start_offset == truncate_offset, f"Start offset: {topic_info.start_offset}"
+        assert topic_info.high_watermark == (
+            num_records + 1), f"High watermark: {topic_info.high_watermark}"
+
+    def delete_records(self, topic, partition, truncate_offset):
+        """
+        Makes delete records call with 1 topic partition in the request body.
+
+        Asserts that the response contains 1 element, with matching topic &
+        partition contents.
+        """
+        response = self.kcl.delete_records(
+            {topic: {
+                partition: truncate_offset
+            }})
+        assert len(response) == 1
+        assert response[0].topic == topic
+        assert response[0].partition == partition
+        assert response[0].error == 'OK', f"Err msg: {response[0].error}"
+        return response[0].new_low_watermark
+
+    def final_segment_boundaries(self, snapshot, leader):
+        # Find the base offset of the final segment
+        def to_base_index(seg):
+            if seg.base_index is not None:
+                return int(seg.base_index.split('-')[0])
+            if seg.data_file is not None:
+                return int(seg.data_file.split('-')[0])
+            return 0
```

Comment:
What is the valid condition where a segment from the python scan of local storage would have no base index?  This feels like more of an error case.